brickBreaker.js

```diff
@@ -0,0 +1,166 @@
+        x += 50;
+        this.bricks[row][col] = brick;
+      }
+      y += 25;
+    }
+  }
+
+  initializeEventListeners() {
+    window.addEventListener("keydown", this.keyDownHandler.bind(this));
+    window.addEventListener("keyup", this.keyUpHandler.bind(this));
+  }
+
+  keyDownHandler(e) {
+    switch (e.key) {
+      case "ArrowLeft":
+        this.paddle.setDirection(-1);
+        break;
+      case "ArrowRight":
+        this.paddle.setDirection(1);
+        break;
+    }
+  }
+
+  keyUpHandler(e) {
+    switch (e.key) {
+      case "ArrowLeft":
+        this.paddle.setDirection(0);
+        break;
+      case "ArrowRight":
+        this.paddle.setDirection(0);
+        break;
+    }
+  }
+
+  updateEntities() {
+    this.paddle.move();
+    this.ball.move();
+  }
+
+  detectPaddleCollision() {
+    if (
+      this.paddle.top <= this.ball.bottom &&
+      this.ball.top >= this.paddle.bottom
+    ) {
+      if (
+        this.paddle.left <= this.ball.left &&
+        this.paddle.right >= this.ball.right
+      ) {
+        this.ball.setDirection(-1, -1);
+      }
+    }
+  }
+
+  detectCanvasCollision() {
+    if (this.ball.left <= 0) {
+      this.ball.setDirection(1, this.ball.yDirection);
+    } else if (this.ball.right >= 400) {
+      this.ball.setDirection(-1, this.ball.yDirection);
+    } else if (this.ball.top <= 0) {
+      this.ball.setDirection(this.ball.xDirection, 1);
+    } else if (this.ball.top > this.canvas.height) {
+      this.ball.setDirection(0, 0);
+    }
+  }
+
+  detectBrickCollision() {
+    for (let row = 0; row < this.bricks.length; row++) {
+      for (let col = 0; col < this.bricks[row].length; col++) {
+        let brick = this.bricks[row][col];
+        if (brick.bottom >= this.ball.top && brick.broken === false) {
+          if (this.ball.right >= brick.left && this.ball.left <= brick.right) {
+            this.ball.setDirection(1, 1);
+            brick.broken = true;
+            this.score += 1;
+          }
+        }
+      }
+    }
+  }
+
+  clearCanvas() {
+    this.ctx.clearRect(0, 0, this.canvas.width, this.canvas.height);
+  }
+
+  reset() {
+    this.clearCanvas();
+    this.initializeEntities();
+  }
+
+  update() {
+    this.updateEntities();
+    this.detectPaddleCollision();
+    this.detectCanvasCollision();
+    this.detectBrickCollision();
+  }
+
+  renderBricks() {
+    for (let row = 0; row < this.bricks.length; row++) {
+      for (let col = 0; col < this.bricks[row].length; col++) {
+        let brick = this.bricks[row][col];
+        if (brick.broken === false) {
+          brick.render();
+        }
+      }
+    }
+  }
+
+  renderScore() {
+    this.ctx.font = "20px Arial";
+    this.ctx.fillText(`Score: ${this.score}`, 300, 575);
+    if (this.score === 40) {
+      this.ctx.font = "30px Arial";
+      this.ctx.fillText(
+        "YOU WIN",
+        this.canvas.width / 2,
+        this.canvas.height / 2
+      );
+      this.reset();
```

Comment:
should reset on a keypress, otherwise you never see this text right? you clear the canvas in reset